proto/src/serializers/bytes.rs

```diff
@@ -0,0 +1,191 @@
+    where
+        D: Deserializer<'de>,
+    {
+        let s = Option::<CowStr>::deserialize(deserializer)?.unwrap_or_default();
+        String::from_utf8(base64::decode(s).map_err(serde::de::Error::custom)?)
+            .map_err(serde::de::Error::custom)
+    }
+
+    /// Serialize from T into base64string
+    pub fn serialize<S, T>(value: &T, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+        T: AsRef<[u8]>,
+    {
+        let base64_bytes = base64::encode(value.as_ref());
+        let base64_string = String::from_utf8(base64_bytes).map_err(serde::ser::Error::custom)?;
+        serializer.serialize_some(&base64_string)
+    }
+}
+
+/// Serialize into and deserialize from a sequence of _base64string_.
+pub mod vec_base64string {
+    use crate::serializers::cow_str::CowStr;
+    use serde::{Deserialize, Deserializer, Serializer};
+    use subtle_encoding::base64;
+
+    /// Deserialize array into `Vec<Vec<u8>>`
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<Vec<Vec<u8>>, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        Option::<Vec<CowStr>>::deserialize(deserializer)?
+            .unwrap_or_default()
+            .into_iter()
+            .map(|s| base64::decode(s).map_err(serde::de::Error::custom))
+            .collect()
+    }
+
+    /// Serialize from `Vec<T>` into `Vec<base64string>`
+    pub fn serialize<S, T>(value: &[T], serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+        T: AsRef<[u8]>,
+    {
+        let base64_strings = value
+            .iter()
+            .map(|v| {
+                String::from_utf8(base64::encode(v.as_ref())).map_err(serde::ser::Error::custom)
+            })
+            .collect::<Result<Vec<String>, S::Error>>()?;
+        serializer.serialize_some(&base64_strings)
+    }
+}
+
+/// Serialize into and deserialize from an optional _base64string_.
+pub mod option_base64string {
+    use crate::serializers::cow_str::CowStr;
+    use serde::{Deserialize, Deserializer, Serializer};
+    use subtle_encoding::base64;
+
+    /// Deserialize `Option<base64string>` into `Vec<u8>` or null
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<Vec<u8>, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = Option::<CowStr>::deserialize(deserializer)?.unwrap_or_default();
+        base64::decode(s).map_err(serde::de::Error::custom)
+    }
+
+    /// Serialize from `T` into `Option<base64string>`
+    pub fn serialize<S, T>(value: &T, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+        T: AsRef<[u8]>,
+    {
+        let base64_bytes = base64::encode(value.as_ref());
+        let base64_string = String::from_utf8(base64_bytes).map_err(serde::ser::Error::custom)?;
+        serializer.serialize_some(&base64_string)
+    }
+}
+
+/// Serialize into string, deserialize from string
+pub mod string {
```

Comment:
please remove this one if unused, it sounds not very practical

Reply:
done